### ISSUE TITLE
units: Remove deprecated functions 

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#3776]
In #3759 we introduced deprecation notices to `units::amount`. These should be removed before we release `v1.0`.

**Comment by tcharding** [comment]
I think you told me this before but is in ok to remove deprecated stuff without doing a new major release? When I wrote this issue I thought not but you comment makes me think this is whats implied?

**Comment by apoelstra** [comment]
>is in ok to remove deprecated stuff without doing a new major release? 

No.

**Comment by tcharding** [comment]
> "1.0 with all the deprecated functions still there"

Oh you mean we do a final round of pre-1.0 releases, do a 0.33.0 release, remove the deprecated stuff, then drop the 1.0s

Don't worry, I get there in the end.

**Comment by jamillambert** [comment]
> Oh you mean we do a final round of pre-1.0 releases of leaf crates, do a bitcoin 0.33.0 release, remove the deprecated stuff, then drop the 1.0s

I like this approach too. That way there is a release with all deprecated stuff in it for those that need it, but everything else up to date.  And then if users want to move to 1.0 and put in the effort to remove reliance on deprecated stuff at least they know it's a stable API.

**Comment by tcharding** [comment]
Lets do it.

> Sorry for my terse reply 

No stress, it made me realise the info must already be there in front of me.

**Comment by tcharding** [comment]
> I'd kinda like to do a 0.33 which is "1.0 with all the deprecated functions still there".

We discussed that we do 0.33-alpha.x which needs leaf crate releases eg `units 1.0-alpha.0`. If in a leaf crate the final `1.0-alpha.x` release is the one that has "everything including deprecated stuff  right before 1.0 release" in it then we need to be mindful that we need a `rust-bitcoin` release that depends on all the leaf crate alphas before we start 1.0'ing any of them (otherwise we loose the deprecated stuff prematurely).

**Comment by apoelstra** [comment]
Yeah, separate from the alpha cycle I'd like to release a 0.33 with new non-alpha versions of leaf crates.

**Comment by tcharding** [comment]
Removed already from `units`.
